ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/api/KeyValueParticipantStateWriter.scala

```diff
@@ -73,14 +76,27 @@ class KeyValueParticipantStateWriter(
       loggingContext: LoggingContext,
       telemetryContext: TelemetryContext,
   ): CompletionStage[SubmissionResult] = {
-    val submission = keyValueSubmission
-      .archivesToSubmission(
-        submissionId,
-        Conversions.archivesToHashesAndBytes(archives),
-        sourceDescription.getOrElse(""),
-        writer.participantId,
-      )
-    commit(submissionId, submission)
+    ArchiveConversions.parsePackageIdsAndRawArchives(archives) match {
+      case Left(_) =>
+        CompletableFuture.completedFuture(
+          SubmissionResult.SynchronousError(
```

Comment:
@fabiotudone-da please take a look if this is fine.

Reply:
Seems sensible to me!